### PR TITLE
Add some tests from records test plan

### PR DESCRIPTION
**Pull request by jcouv**
Relates to https://github.com/dotnet/roslyn/issues/40726 (test plan for records)

### CODE REVIEW

File: src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs [hunk 1]
```diff
@@ -428,6 +508,44 @@ partial class C
...
+{
+    public int P1 { get; set; } = X;
+}
+public partial record C
+{
+    public int P2 { get; set; } = X;
+}
+";
+            var comp = CreateCompilation(new[] { src, IsExternalInitTypeDefinition }, options: TestOptions.DebugExe, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics();
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs:528]: >.VerifyDiagnostics() [](start = 16, length = 20)

Consider calling VarifyDiagnostics on result of CompileAndVerify. That set will include any warnings that might be reported by emit phase.

File: src/Compilers/CSharp/Test/Symbol/Symbols/Source/RecordTests.cs [hunk 2]
```diff
@@ -1176,6 +1176,21 @@ .maxstack  2
...
+        [Fact]
+        public void PartialTypes_04_PartialBeforeModifiers()
+        {
+            var src = @"
+partial public record C
+{
+}
+";
+            CreateCompilation(src).VerifyDiagnostics(
+                // (2,1): error CS0267: The 'partial' modifier can only appear immediately before 'class', 'struct', 'interface', or 'void'
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Symbol/Symbols/Source/RecordTests.cs:1188]: >before 'class', 'struct', 'interface', or 'void' [](start = 91, length = 48)

Is this list adjusted in another PR?

Comment (jcouv) [reply src/Compilers/CSharp/Test/Symbol/Symbols/Source/RecordTests.cs]: Yes. I'm merged master into this PR and updated the comment now.

**Comment by jcouv** [comment]
Since this PR contains no product code change (test and updating a resource string), I'll go ahead and merge with a single review once CI is green. Thanks

**Pull request merged**